src/utils/include/utils/Array.hpp

```diff
@@ -125,6 +125,14 @@ template <typename T, std::size_t N> struct Array {
     return ret;
   }
 
+#ifdef __HCC__
+  // workaround for https://github.com/RadeonOpenCompute/hcc/issues/860
+  __attribute__((annotate("serialize"))) void
```

Comment:
@RudolfWeeber can you explain what these lines do?

Reply:
They just force the compiler into a certain overload resolution. Function prototypes are sufficient since we don't actually need to serialize data.